target/linux/ath79/image/generic.mk

```diff
@@ -50,6 +50,25 @@ define Build/cybertan-trx
 	-rm $@-empty.bin
 endef
 
+define Build/edimax-headers
+	$(eval edimax_magic=$(word 1,$(1)))
+	$(eval edimax_model=$(word 2,$(1)))
+
+	$(STAGING_DIR_HOST)/bin/edimax_fw_header -M $(edimax_magic) -m $(edimax_model)\
+		-v $(VERSION_DIST)$(firstword $(subst +, , $(firstword $(subst -, ,$(REVISION))))) \
+		-n "uImage" \
+		-i $(KDIR)/loader-$(DEVICE_NAME).uImage \
+		-o $@.uImage
+	$(STAGING_DIR_HOST)/bin/edimax_fw_header -M $(edimax_magic) -m $(edimax_model)\
+		-v $(VERSION_DIST)$(firstword $(subst +, , $(firstword $(subst -, ,$(REVISION))))) \
+		-n "rootfs" \
+		-i $@ \
+		-o $@.rootfs
+	cat $@.uImage $@.rootfs > $@.new
+	mv $@.new $@
```

Comment:
Ah, haven't been aware of this updated version. The appeal of this approach actually is that we don't need $@.new at all, because we should be able to directly
`cat $@.uImage $@.rootfs > $@`
and drop the move ...

Reply:
Good idea, will do that